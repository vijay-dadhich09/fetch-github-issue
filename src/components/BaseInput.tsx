import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

const BaseInput = ({ children, label }) => (
  <View style={styles.baseInput}>
    <Text style={styles.text}>{label}</Text>
    {children}
  </View>
);

const styles = StyleSheet.create({
  baseInput: {
    width: '80%',
    paddingVertical: 6,
  },
  text: {
		color: '#273746',
		fontSize: 15,
		fontWeight: "600",
	}
});

export default BaseInput;