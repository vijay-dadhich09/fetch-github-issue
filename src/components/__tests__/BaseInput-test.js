import React from 'react';
import { TextInput } from 'react-native';
import { render } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import BaseInput from '../BaseInput';

describe('Header test', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <BaseInput label="Enter Author name">
          <TextInput
            accessibilityLabel="input values"
            style={{}}
            onChangeText={jest.fn()}
            value=""
            autoCapitalize="none"
          />
        </BaseInput>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders Labels', () => {
    const { getByText } = render(
      <BaseInput label="Enter Author name">
      <TextInput
        accessibilityLabel="input values"
        style={{}}
        onChangeText={jest.fn()}
        value=""
        autoCapitalize="none"
      />
    </BaseInput>
    );
    expect(getByText('Enter Author name')).toHaveTextContent('Enter Author name');
  });
});
