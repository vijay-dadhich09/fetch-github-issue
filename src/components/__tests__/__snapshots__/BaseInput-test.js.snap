// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test renders correctly 1`] = `
<View
  style={
    Object {
      "paddingVertical": 6,
      "width": "80%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#273746",
        "fontSize": 15,
        "fontWeight": "600",
      }
    }
  >
    Enter Author name
  </Text>
  <TextInput
    accessibilityLabel="input values"
    allowFontScaling={true}
    autoCapitalize="none"
    onChangeText={[MockFunction]}
    rejectResponderTermination={true}
    style={Object {}}
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;
